ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/messages/TransactionsMessage.java

```diff
@@ -47,7 +49,24 @@ public static TransactionsMessage create(final Iterable<Transaction> transaction
     return new TransactionsMessage(tmp.encoded());
   }
 
-  private TransactionsMessage(final BytesValue data) {
+  public static LimitedTransactionsMessages createLimited(
+      final Iterable<Transaction> transactions, final TransactionsMessageLimiter limiter) {
+    final Set<Transaction> includedTransactions = new HashSet<>();
+    final BytesValueRLPOutput tmp = new BytesValueRLPOutput();
+    tmp.startList();
+    for (final Transaction transaction : transactions) {
+      transaction.writeTo(tmp);
+      includedTransactions.add(transaction);
+      if (limiter.isLastTransactionToIncludeWithoutExceedingLimit(tmp)) {
+        break;
```

Comment:
It looks like the logic here is packing up transactions until we exceed the 1MB limit.  There's no check that the last transaction isn't huge such that the whole message could exceed the hard 16MB limit.

Reply:
We create a follow up ticket to implement this check at lower level (devp2p layer)